src/memory/memory.h

```diff
@@ -46,6 +46,12 @@ typedef enum {
     MEMORY_ERR_UNKNOWN,
 } memory_result_t;
 
+typedef enum {
+    MEMORY_AREA_SHARED_DATA,
+    MEMORY_AREA_APP_DATA,
+    MEMORY_AREA_SMARTEEPROM,
+} memory_area_t;
```

Comment:
it is also used in mock_memory.h/c.

Will add comments

Reply:
ah, right. and forward decl. of enums is forbidden in ISO C. that's a shame. 
hold on. why is this an enum, why not do something like this:

```c
// area must be FLASH_SHARED_DATA_START, FLASH_APPDATA_START or eeprom.
static bool _write_to_address(uint32_t area, uint32_t addr, uint8_t* chunk)
```

it would still be easy to branch logic in tests and no need to declare unused types in headers.

Reply:
well they are not unused lol.

i felt like switching on an enum was clearer to read than switching on an actual address value :shrug: 

i can also remove the enum and switch on actual the address values in the tests. let me try that..

Reply:
i removed the enum and use only the base address now, PTAL

Reply:
(had a rebase error, now should be good)

Reply:
:+1: seems cleaner to me and less code.
I think you might be subconsciously trying to make it look more like rust but C is not rust :)